lib/ansible/modules/cloud/amazon/ec2_asg.py

```diff
@@ -952,9 +952,15 @@ def create_autoscaling_group(connection, module):
                     attach_lb_target_groups(connection, group_name, list(tgs_to_attach))
 
         # check for attributes that aren't required for updating an existing ASG
-        desired_capacity = desired_capacity if desired_capacity is not None else as_group['DesiredCapacity']
-        min_size = min_size if min_size is not None else as_group['MinSize']
-        max_size = max_size if max_size is not None else as_group['MaxSize']
+        # check if min_size/max_size/desired capacity have been specified and if not use ASG values
+        if min_size is None:
+            min_size = as_group['MinSize']
+        if max_size is None:
+            max_size = as_group['MaxSize']
+        if desired_capacity is None and as_group['DesiredCapacity'] is None:
```

Comment:
Prefer:

```
if desired_capacity is None:
    if as_group['DesiredCapacity'] is None:
        desired_capacity = min_size
    else:
        desired_capacity = as_group['DesiredCapacity']
```

Reply:
Yes, that does read better. I'll update it.